src/material.rs

```diff
@@ -0,0 +1,353 @@
+//! Material module.
+//!
+//! Provides:
+//!  * Different pigments that implement [`GetColor`] trait ;
+//!  * Different BRDF that implement both [`Eval`] and [`ScatterRay`] trait;
+//!  * A [`Material`] thanks to pigments and BRDF.
+use crate::color::{Color, BLACK, WHITE};
+use crate::hdrimage::HdrImage;
+use crate::misc::Vector2D;
+use crate::normal::{create_onb_from_z, Normal};
+use crate::point::Point;
+use crate::random::Pcg;
+use crate::ray::Ray;
+use crate::vector::Vector;
+use std::f32::consts::PI;
+
+/// Trait that associates a [`Color`] with each point on a parametric surface `(u,v)`.
+pub trait GetColor {
+    fn get_color(&self, uv: Vector2D) -> Color;
+}
+
+/// A uniform pigment.
+///
+/// This is the most boring pigment: a uniform hue over the whole surface.
+#[derive(Default)]
```

Comment:
Deriving `Debug` is always a good idea, also i think we could derive `Copy` and `Clone` since `Color` has them.